File: swing/src/main/java/wannabe/swing/renderer/RoundedSquare.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.swing.SwingProjected;

public class RoundedSquare extends SwingRenderer {
  @Override public void draw(Graphics g, SwingProjected p) {
    int arc = p.size / 3;
    g.drawRoundRect(p.left, p.top, p.size, p.size, arc, arc);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/RoundedSquare.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class RoundedSquare : SwingRenderer() {
  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    val arc = p.size / 3
    g.drawRoundRect(p.left, p.top, p.size, p.size, arc, arc)
  }
}
